### PR TITLE
fix: accurate loss reporting

### PR DESCRIPTION
**Pull request by rjwalters**
### Problem

We have been reporting incremental averages during loss accumulation:

```python
self.losses.policy_loss += pg_loss.item() / total_minibatches
```

This approach will lead to **Incorrect averages on early exit**: 

When training exits early due to KL divergence threshold, we've divided by total_minibatches but only accumulated a subset

### Solution

This PR changes the loss tracking to:
1. Accumulate raw loss values without division
2. Track the actual number of minibatches processed
4. Compute averages only when needed in `to_dict()`

### Changes

- Modified `Losses` class to store cumulative sums (`*_sum` attributes)
- Added `minibatches_processed` counter
- Updated `to_dict()` to compute true averages: `sum / actual_minibatches_processed`
- Fixed KL divergence check to use the running average instead of the last minibatch value

    

---

## Summary by cubic
Fixed loss averaging to ensure accurate training stats and correct KL-based early stopping, even when training exits early.

- **Bug Fixes**
  - Loss values now accumulate raw sums and track the number of minibatches processed.
  - Averages are computed only when needed, using actual minibatch counts.
  - KL divergence check now uses the running average, not just the last minibatch.

### CODE REVIEW

**Comment by dominik3141** [comment]
Have you done some empirical testing? How much did the stats change?

**Comment by rjwalters** [comment]
> Have you done some empirical testing? How much did the stats change?

image

No - but I don't think we should expect to see a difference unless we are exiting early via

```

            if self.trainer_cfg.target_kl is not None:
                average_approx_kl = self.losses.approx_kl_sum / self.losses.minibatches_processed
                if average_approx_kl > self.trainer_cfg.target_kl:
                    break
                    
```
